tests/integration/api/test_xai.py

```diff
@@ -95,6 +95,9 @@ def test_predict_with_explain(
     if "dino" in model_name or "rtmdet_inst_tiny" in model_name:
         pytest.skip("DINO and Rtmdet_tiny are not supported.")
 
+    if "mobilenet_v3_large" in model_name:
+        pytest.skip("There's issue with mobilenet_v3_large model. Skip for now.")
```

Comment:
Hi, @sovrasov @negvet  When I run the code below on this PR, the outputs of exported_model with XAI is weird (feature_vector & silency_map shape), Is there something I'm missing?
```python
from pathlib import Path

import openvino.runtime as ov
from otx.engine import Engine

recipe = "src/otx/recipe/classification/multi_class_cls/mobilenet_v3_large.yaml"
data_root = "tests/assets/classification_dataset"
work_dir = "otx-workspace-xai-test"

engine = Engine.from_config(
    config_path=recipe,
    data_root=data_root,
    work_dir=work_dir,
)


predict_result_explain_torch = engine.predict(explain=True)
ckpt_path = Path(work_dir) / "checkpoint.ckpt"
engine.trainer.save_checkpoint(ckpt_path)
exported_model_path = engine.export(checkpoint=ckpt_path, explain=True)


model = ov.Core().read_model(exported_model_path)
feature_vector_output = None
saliency_map_output = None
for output in model.outputs:
    if "feature_vector" in output.get_names():
        feature_vector_output = output
    if "saliency_map" in output.get_names():
        saliency_map_output = output
```

![image](https://github.com/openvinotoolkit/training_extensions/assets/38045080/909d82f3-95d3-42d1-9842-350358afd50e)
in ov model api side,
![image](https://github.com/openvinotoolkit/training_extensions/assets/38045080/d11b0664-bd2e-455c-81aa-8823b2e736e0)

Reply:
Looks like output names for XAI are not aligned correctly with outputs.
Based on shapes names[965] should be `saliency map` and names[885] should be the `feature vector`. 
Probably it happened after latest export/XAI refactoring. I'll create a ticket for this issue and investigate it.

Thank you for noticing and reporting this behavior!